### PR TITLE
Move the definition of `__thefunc__` and add AUTO_TRACE

### PR DESCRIPTION
**Pull request by d7919**
Moved definition of `__thefunc__` into msg_stack and adds `AUTO_TRACE` macro that works like `TRACE` but automatically generates the recorded string using `__thefunc__`.

### CODE REVIEW

**Comment by d7919** [comment]
Originally `__thefunc__` was define in `scorepwrapper.hxx` but it seems more appropriate to separate this from the profiling header.

**Comment by dschwoerer** [comment]
Is there a reason `AUTO_TRACE` ignores arguments?
If I add arguments, I would expect them to be included, thus not getting an error seems unexpected.

File: include/msg_stack.hxx [hunk 1]
```diff
@@ -172,4 +184,6 @@ private:
 #define TRACE(...)
 #endif
 
+#define AUTO_TRACE(...) TRACE("%s", __thefunc__)
```
Comment (ZedThree) [inline include/msg_stack.hxx]: Please could you add a doxygen comment?

**Comment by d7919** [comment]
@dschwoerer and @ZedThree  I think I've now addressed your comments.

File: include/msg_stack.hxx [hunk 2]
```diff
@@ -172,4 +184,23 @@ private:
...
+ *
+ * Example
+ * -------
+ *
+ * {
+ *   AUTO_TRACE();
+ *
+ * } // Scope ends, message popped
+ */
+#define AUTO_TRACE()("%s", __thefunc__)
```
Comment (bendudson) [inline include/msg_stack.hxx]: I think this is missing a `TRACE` macro call

**Pull request merged**